### ISSUE TITLE
Empty ids in generated symbols

### ISSUE THREAD

**Issue by charlestati** [svg-sprite/svg-sprite#242]
Hello!

When I generate a sprite in `symbol` mode, all the symbols' ids are empty.

When I use a custom generator with a callback I am able to set an id. However, the parameter of the callback is an empty string which means I cannot build an id based on the file name.

Any idea of what I am doing wrong? I am using **svg-sprite** version **1.3.7**

Here is my code:

```javascript
const path = require('path')
const async = require('async')
const SVGSpriter = require('svg-sprite')

const config = require('../config')

function build() {
  const spriterConfig = {
    mode: {
      symbol: {
        inline: true,
      },
    },
  }

  const spriter = new SVGSpriter(spriterConfig)

  logInfo('Building SVG sprite')

  readDirectory(config.svg.inputDir).then(files => {
    async.each(
      files,
      (file, done) => {
        const filepath = path.join(config.svg.inputDir, file)
        readFile(filepath)
          .then(data => {
            spriter.add(filepath, file, data) // I asynchronously add my files here
            done()
          })
          .catch(error => done(error))
      },
      error => {
        if (error) {
          logError(error.toString())
        } else {
          spriter.compile((error, result) => {
            const sprite = result.symbol.sprite
            writeFile(
              path.join(config.svg.outputDir, 'sprite.svg'),
              sprite.contents
            ).then(() => logSuccess('SVG sprite built'))
          })
        }
      }
    )
  })
}
```

Thank you and have a great day!

**Comment by MadLittleMods** [comment]
@charlestati See https://github.com/jkphl/svg-sprite/issues/112

I was running into the same thing but the keys were to use full absolute URLs when using `spriter.add(...)`

**Comment by Kreeg** [comment]
@MadLittleMods @charlestati this should have been fixed in #325 Can you confirm?
